### DISCUSSION TITLE
OSQuery - Mac Install

### DISCUSSION THREAD

**Discussion by secraven** [Security-Onion-Solutions/securityonion#1851] [Unsupported Versions] [answered]
I am having a problem getting my Mac to report into SO.  I downloaded the package from SO and also tried the package directly from OSQuery.  I followed the directions below.  Could someone give me so pointers on how to get my Mac connected? I have a Windows system setup already using the provided packages.

Thanks!

https://osquery.readthedocs.io/en/stable/installation/install-macos/

**Comment by secraven** [comment, marked as answer]
Has anyone got OSQuery working successfully on a Mac?

**Reply by weslambert** [reply]
Did you notice any errors on the server side or client side that could indicate why it won't connect?

**Reply by tommyluke** [reply]
I've not been successful. The https://docs.securityonion.net/en/2.3/osquery.html#osquery documentations states "The macOS package is a stock Launcher package, and will require additional configuration once it has been deployed." <-- where can I find information about the additional config?

**Reply by defensivedepth** [reply]
@secraven @tommyluke I just updated our docs to show what additional config is needed:

https://docs.securityonion.net/en/2.3/osquery.html#agents-deployment

It's a bit of extra work, which is why we build the packages for the other platforms. Still looking at ideas to do macOS packaging better.

**Reply by tommyluke** [reply]
@defensivedepth This works well, thank you so much for the help.

**Reply by cowbe0x004** [reply]
Do I need to install osquery as outlined in osquery doc? Or just run the pkg-launcher.pkg? I ran pkg-launcher.pkg and there's osqueryd in /usr/local/so-launcher/bin/. 
SO doc says to restart osquery agent if it's not connecting to fleet, how do I do that? launcherctl list shows there's com.so-launcher.launcher but I don't know how to use it. I'm new to Mac. Thank you.

EDIT: to restart the launcher.
`sudo launchctl kickstart -k system/com.so-launcher.launcher`

**Reply by defensivedepth** [reply]
@cowbe0x004 Sounds like you got it figured out?

**Reply by cowbe0x004** [reply]
Yes thank you. The pkg-launcher.pkg doesn't install osqueryi so I thought I needed to do something else. Looks like just the daemon is enough.

**Reply by defensivedepth** [reply]
osqueryi and osqueryd are actually the same binary - if you run `osqueryd --S ` from the command line, you will drop into the osquery shell (ie osqueryi)
